src/pages/references/node-configuration.md

```diff
@@ -0,0 +1,489 @@
+```
+
+#### p2p_bind
+
+Address and port stacks-node should bind to for P2P connections.
+
+Example:
+
+```toml
+p2p_bind = "0.0.0.0:20444"
+```
+
+#### data_url (optional)
+
+Address and port from which the stacks-node will be receiving incoming rpc connections.
+
+Example:
+
+```toml
+data_url = "1.2.3.4:20443"
+```
+
+#### p2p_address (optional)
+
+Address and port from which the stacks-node will be receiving incoming p2p connections.
+
+Example:
+
+```toml
+p2p_address = "1.2.3.4:20444"
+```
+
+#### bootstrap_node (optional)
+
+Public key, address, and port stacks-node should use to pull transaction data from when starting.
+
+Example:
+
+```toml
+bootstrap_node = "048dd4f26101715853533dee005f0915375854fd5be73405f679c1917a5d4d16aaaf3c4c0d7a9c132a36b8c5fe1287f07dad8c910174d789eb24bdfb5ae26f5f27@krypton.blockstack.org:20444"
+```
+
+#### wait_time_for_microblocks (optional)
+
+Time to wait (in milliseconds) for microblocks.
+
+Example:
+
+```toml
+wait_time_for_microblocks = 15000
+```
+
+#### seed (optional)
+
+The private key to use for mining. Only needed if `miner` is set to `true`.
+
+[See this page for information on how to generate a private key.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+seed = "replace-with-your-private-key"
+```
+
+#### local_peer_seed (optional)
+
+The private key to use for mining.
+
+[See this page for information on how to generate a private key.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+local_peer_seed = "replace-with-your-private-key"
+```
+
+#### miner (optional)
+
+Determines whether the stacks-node is running a follower (`false`) or a miner (`true`).
+
+[See this page for information on how to run a miner.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+miner = true
+```
+
+#### mine_microblocks (optional)
+
+Determines whether the stacks-node will mine microblocks. Will only take effect if `miner` is set to `true`.
+
+[See this page for information on how to run a miner.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+mine_microblocks = true
+```
+
+#### prometheus_bind (optional)
+
+Address and port stacks-node should open for Prometheus metrics collection.
+
+Example:
+
+```toml
+prometheus_bind = "0.0.0.0:9153"
+```
+
+### Section: events_observer (optional)
+
+Contains options for watching events emitted by a local [stacks-blockchain-api](https://github.com/blockstack/stacks-blockchain-api) service.
+
+-> This section can be repeated multiple times.
+
+Example:
+
+```toml
+[[events_observer]]
+endpoint = "address-to-my-local.stacks-node-api.com:3700"
+retry_count = 255
+events_keys = ["*"]
+```
+
+#### endpoint
+
+Address and port to a stacks-node-api to watch for events.
+
+Example:
+
+```toml
+endpoint = "address-to-my-local.stacks-node-api.com:3700"
+```
+
+#### retry_count
+
+Number of times to retry before failing.
+
+Example:
+
+```toml
+retry_count = 255
+```
+
+#### events_keys
+
+Event keys for which to watch.
```

Comment:
The emitted node events can be restricted by account, function name and event type. Asterix ("*") can be used to emit all events.